manifests/vrouter.pp

```diff
@@ -207,19 +210,48 @@
     unless => "/sbin/ifconfig | grep ^${vrouter_physical_interface}",
   }
 
-  network_config { $vrouter_interface:
-    ensure  => present,
-    family  => 'inet',
-    method  => 'dhcp',
-    onboot  => true,
-    options => {
-                'pre-up' => "${vrouter_patch}/usr/local/bin/if-vhost0",
-                },
-  } ->
-  exec { "ifup_${vrouter_interface}":
-    command => "/sbin/ifup ${vrouter_interface}",
-    unless  => "/sbin/ifconfig | grep ^${vrouter_interface}",
-    require => Package[$package_names],
+  if $interface_is_dhcp {
+    network_config { $vrouter_interface:
+      ensure  => present,
+      family  => 'inet',
+      method  => 'dhcp',
+      onboot  => true,
+      options => {
+                  'pre-up' => "${vrouter_patch}/usr/local/bin/if-vhost0",
+                  },
+    } ->
+    exec { "ifup_${vrouter_interface}":
+      command => "/sbin/ifup ${vrouter_interface}",
+      unless  => "/sbin/ifconfig | grep ^${vrouter_interface}",
+      require => Package[$package_names],
+    }
+  }
+  else {
+    package {'ifupdown-extra':}
+    network_config { $vrouter_interface:
+      ensure  => present,
+      family  => 'inet',
+      method  => 'static',
+      ipaddress => $vrouter_ip,
+      netmask  => $vrouter_netmask,
+      onboot  => true,
+      options => {
+                  'pre-up' => "${vrouter_patch}/usr/local/bin/if-vhost0",
+                  },
+    } ->
+    network_route { 'default':
+      ensure => present,
+      gateway => $vrouter_gw_orig,
+      interface => $vrouter_interface,
+      netmask => '0.0.0.0',
+      network => 'default',
+      require => Package['ifupdown-extra']
+    } ->
+    exec { "ifup_${vrouter_interface}":
```

Comment:
Let me check.. i think i can put the exec out of the if else block and then add a require condition. Since this already running for now we will merge it